File: wrong_code/IsValidBST.java
import java.util.List;
import java.util.Arrays;
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int val) {
        this.val = val;
    }
}

public class IsValidBST {
    
    public static boolean isValidBST(TreeNode root, List<Integer> vec) {
        // 空树的情况
        if (root == null) {
            return true;
        }

        // 检查 vec 是否严格递增
        for (int i = 1; i < vec.size(); i++) {
            if (vec.get(i) < vec.get(i - 1)) { // 错误：应该是 <=
                return false;
            }
        }
        return true;
    }

    

    public static void main(String[] args) {
        // 示例二叉树
        TreeNode root = new TreeNode(2);
        root.left = new TreeNode(1);
        root.right = new TreeNode(3);

        // 模拟生成 vec
        List<Integer> vec = Arrays.asList(1, 2, 3);

        // 验证 BST
        boolean result = isValidBST(root, vec);
        System.out.println("Is valid BST: " + result);
    }
}

File: hardToTest/IsValidBST/wrong_code/IsValidBST.py
def IsValidBST( root,vec):
    #self.vec = []  # 清空数组
    if not root:
        return True
    
    for i in range(1, len(vec)):
        # 注意要小于等于，搜索树里不能有相同元素
        if vec[i] < vec[i - 1]: #wrong code
            return False
    return True
if __name__ == "__main__":
    root = [2,1,3]
    vec = [2,1,3]
    print(IsValidBST(root,vec))